RecoLocalCalo/EcalRecProducers/plugins/DeclsForKernelsPh2WeightsGPU.h

```diff
@@ -0,0 +1,71 @@
+#ifndef RecoLocalCalo_EcalRecProducers_plugins_DeclsForKernels_h
+#define RecoLocalCalo_EcalRecProducers_plugins_DeclsForKernels_h
+
+#include <vector>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+#include "CUDADataFormats/EcalDigi/interface/DigisCollection.h"
+#include "CUDADataFormats/EcalRecHitSoA/interface/EcalRecHit.h"
+#include "CUDADataFormats/EcalRecHitSoA/interface/EcalUncalibratedRecHit.h"
+#include "CUDADataFormats/EcalRecHitSoA/interface/RecoTypes.h"
+#include "CondFormats/EcalObjects/interface/EcalChannelStatus.h"
+#include "CondFormats/EcalObjects/interface/EcalChannelStatusCode.h"
+#include "CondFormats/EcalObjects/interface/EcalGainRatios.h"
+#include "CondFormats/EcalObjects/interface/EcalGainRatiosGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalIntercalibConstantsGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalLaserAPDPNRatiosGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalLaserAPDPNRatiosRefGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalLaserAlphasGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalLinearCorrectionsGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalMultifitParametersGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalPedestals.h"
+#include "CondFormats/EcalObjects/interface/EcalPedestalsGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalPulseCovariancesGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalPulseShapesGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalRechitADCToGeVConstantGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalRechitChannelStatusGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalSamplesCorrelationGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalTimeBiasCorrections.h"
+#include "CondFormats/EcalObjects/interface/EcalTimeBiasCorrectionsGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalTimeCalibConstantsGPU.h"
+#include "CondFormats/EcalObjects/interface/EcalTimeOffsetConstant.h"
+#include "CondFormats/EcalObjects/interface/EcalWeightSet.h"
+#include "HeterogeneousCore/CUDAUtilities/interface/cudaCheck.h"
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+
+#include "EigenMatrixTypes_gpu.h"
+
+struct EcalPulseShape;
+class EcalSampleMask;
+class EcalTimeBiasCorrections;
+struct EcalPulseCovariance;
+class EcalDigiCollection;
+class EcalXtalGroupId;
+class EcalSamplesCorrelation;
+class EBDigiCollection;
+class EEDigiCollection;
+
+namespace ecal {
+  namespace weights {
+
+    struct EventOutputDataGPUWeights {
```

Comment:
```suggestion
    struct EventOutputDataGPU {
```
Since the name space makes it clear that this is the weights based algo version